File: internal/interfaces/menu/menu_service.go
package interfacemenu

import (
	domainmenu "vendor-management-system/internal/domain/menu"
	"vendor-management-system/internal/dto"
	"vendor-management-system/pkg/filter"
)

type ServiceMenuInterface interface {
	Create(req dto.MenuCreate) (domainmenu.MenuItem, error)
	GetByID(id string) (domainmenu.MenuItem, error)
	GetAll(params filter.BaseParams) ([]domainmenu.MenuItem, int64, error)
	GetActiveMenus() ([]domainmenu.MenuItem, error)
	GetUserMenus(userId string) ([]domainmenu.MenuItem, error)
	Update(id string, req dto.MenuUpdate) (domainmenu.MenuItem, error)
	Delete(id string) error
}

File: frontend/src/services/menuService.ts
import api from '../api/axios';

const menuService = {
  getAll: (params = {}) => api.get('/api/menus', { params }),
  getById: (id: string) => api.get(`/api/menu/${id}`),
  create: (data: any) => api.post('/api/menu', data),
  update: (id: string, data: any) => api.put(`/api/menu/${id}`, data),
  delete: (id: string) => api.delete(`/api/menu/${id}`),
  getActiveMenus: () => api.get('/api/menus/active'),
  getUserMenus: () => {
    console.log('Calling getUserMenus API...');
    console.log('Token:', localStorage.getItem('token'));
    return api.get('/api/menus/me');
  },
};

export default menuService;
